tasks/agent-macos.yml

```diff
@@ -0,0 +1,175 @@
+---
+- name: 'Load user data into JSON string {"uid": <number>, "gid": <number>}'
+  shell:
+    cmd: "dscacheutil -q user -a name {{ ansible_user }} | awk 'BEGIN { RS=\"\\n\"; ORS=\" \" } /uid:/ { print \"{ \\\"uid\\\": \" $2\",\" } /gid:/ { print \"\\\"gid\\\": \" $2 \" }\"}'"
+    executable: /bin/bash
+  changed_when: false
+  register: macos_user_output
+
+- name: Extract JSON user data as variable object
+  set_fact:
+    macos_user_data: "{{ macos_user_output.stdout }}"
```

Comment:
this is maybe a bit obscure - but since the string inside `macos_user_output.stdout` is valid json (i format it that way) ansible will auto parse and assign the object to the variable

Reply:
Could you add a comment that explains this? I think it's ok to do like this, but deserves the explanation.

Reply:
yes, you are right, more detail is needed in context - i've added supporting comments